packages/website/src/components/page-templates/package-example.js

```diff
@@ -58,63 +78,127 @@ const ExampleHeading = styled.h2`
   margin-bottom: ${math.multiply(gridSize, 4)}px;
 `;
 
-const CodeStyle = styled.pre`
-  background-color: rgb(23, 43, 77);
-  box-sizing: border-box;
-  color: ${themed({ light: colors.N60, dark: colors.N60 })};
-  display: block;
-  top: 0;
-  bottom: 0;
-  right: 0;
-  position: fixed;
-  padding: ${gridSize}px;
-  margin: 0;
-  width: ${math.multiply(gridSize, 80)}px;
-  overflow-x: auto;
-  overflow-y: scroll;
+const CodeViewButton = ({ isCodeViewExpanded, handleClick }) => (
+  <div
+    style={{ position: 'absolute', right: '0', top: '100px', zIndex: '500' }}
+  >
+    <Button hitAreaSize="small" onClick={handleClick}>
+      {isCodeViewExpanded ? (
+        <ChevronRightCircleIcon
+          label="collapse"
+          primaryColor={colors.B200}
+          size="large"
+        />
+      ) : (
+        <ChevronLeftCircleIcon
+          label="expand"
+          primaryColor={colors.B200}
+          size="large"
+        />
+      )}
+    </Button>
+  </div>
+);
 
-  & code {
-    font-family: Monaco, Menlo, monospace;
-  }
-`;
+CodeViewButton.propTypes = {
+  isCodeViewExpanded: PropTypes.bool.isRequired,
+  handleClick: PropTypes.func.isRequired,
+};
+
+class PackageExample extends React.Component {
+  state = {
+    showCodeViewButton: true,
+    isCodeViewExpanded: true,
+    shouldExpandCodeWithWideWindow: true,
```

Comment:
Just for funs, we're using react 16.8 (the one with hooks) - it'd be nice to start writing some of these components using a hooks architecture for state management over a class component.

Happy to pair with you on changing this over if you want to. (I also need to gain experience using hooks)

Reply:
> No blocking changes, though I would be keen to do the hooks refactor as a fun Friday afternoon activity.

that sounds good